Classes/GarbageCollector.php

```diff
@@ -155,6 +155,20 @@ public function processDatamap_afterDatabaseOperations($status, $table, $uid, ar
 			|| ($table == 'pages' && !$this->isIndexablePageType($record))
 		) {
 			$this->collectGarbage($table, $uid);
+
+			// if the start time is in the future and not otherwise invisible,
+			// it is necessary to add the item to the queue again to be indexed once the start time is reached
+			if ($this->isStartTimeInFuture($table, $record)
```

Comment:
This will result in the item to be re-added to the index queue event if it was e.g. set to hidden.
Not sure though if that would be a problem?